doc/decisions/iterators.md

```diff
@@ -0,0 +1,52 @@
+# Iterators
+
+## Problem
+
+The internal iterator inside KeySets seems to cause more problems than it solves.
```

Comment:
Maybe write a bit about the background why it was introduced and that this is not used anymore (like discussed in the meeting: to remember errorKey in kdbSet).